src/components/expression/expression-panel.js

```diff
@@ -224,16 +218,66 @@ genotet.ExpressionPanel.prototype.updateGenes = function(gene) {
       text: gene
     };
   });
-
   var profile = this.container.find('#profile select').empty();
-  this.selectProfiles_ = profile.select2({
-    data: genes,
-    multiple: true
-  });
-  this.container.find('#profile .select2-container').css({
-    width: '100%'
-  });
-  this.updateGeneSelections();
+
+  if (!genes.length) {
+    return;
+  }
+
+  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
```

Comment:
You can require it once, wait for the async require to be done, and then
use CustomData later on. This is like registering a service and a same
service won't need to be registered twice. I don't see why a same require
need to be performed multiple times.

If you are **100% sure** that this require needs to be done every time you
use it, then it's actually weird antipattern and you can send me the
documentation. I will investigate more closely.

On Saturday, March 19, 2016, Liana Yang <notifications@github.com> wrote:

> In src/components/expression/expression-panel.js
> <https://github.com/ViDA-NYU/genotet/pull/151#discussion_r56758457>:
>
> >    var profile = this.container.find('#profile select').empty();
> > -  this.selectProfiles_ = profile.select2({
> > -    data: genes,
> > -    multiple: true
> > -  });
> > -  this.container.find('#profile .select2-container').css({
> > -    width: '100%'
> > -  });
> > -  this.updateGeneSelections();
> > +
> > +  if (!genes.length) {
> > +    return;
> > +  }
> > +
> > +  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
>
> According to the document, the require is asynchronously and CustomData
> should be used under the require. And we should use CustomData every time
> we updateGenes, so that we have to require this every time consistently.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/151/files/90437f8ff78facbc443fa2717a721540a82c9157#r56758457>
>